cli/src/program.rs

```diff
@@ -2387,6 +2456,100 @@ fn process_extend_program(
         }))
 }
 
+fn process_migrate_program(
+    rpc_client: &RpcClient,
+    config: &CliConfig,
+    program_pubkey: Pubkey,
+    authority_signer_index: SignerIndex,
+    compute_unit_price: Option<u64>,
+) -> ProcessResult {
+    let payer_pubkey = config.signers[0].pubkey();
```

Comment:
Aren't all these checks already performed in the instruction implementation?

Reply:
Yes, but this gives a bit better error messages before paying any fees. It is also done in all the other instructions.